src/tools/rbd/action/Kernel.cc

```diff
@@ -548,13 +548,61 @@ int execute_map(const po::variables_map &vm,
   utils::normalize_pool_name(&pool_name);
 
   MapOptions default_map_options;
-  r = parse_map_options(
-      g_conf().get_val<std::string>("rbd_default_map_options"),
-      &default_map_options);
+  std::string default_map_key = "rbd_default_map_options";
+
+  librados::IoCtx ioctx;
+  librbd::Image image;
+  r = utils::init_io_ctx(rados, pool_name, nspace_name, &ioctx);
+  if (r < 0) {
+    return r;
+  }
+
+  r = utils::open_image(ioctx, image_name, true, &image);
   if (r < 0) {
-    std::cerr << "rbd: couldn't parse default map options" << std::endl;
     return r;
   }
+
+  std::vector<librbd::config_option_t> options;
+  image.config_list(&options);
+  for (const auto &option : options) {
+    if (option.name == default_map_key){
+      r = parse_map_options(option.value, &default_map_options);
+
+      if (r < 0) {
+        std::cerr << "rbd: couldn't parse image map options" << std::endl;
+        return r;
+      }
+
+      break;
+    }
+  }
+
+  if (default_map_options.size() == 0) {
+    librbd::RBD rbd;
+    for (const auto &option : options) {
+      if (option.name == default_map_key) {
+        r = parse_map_options(option.value, &default_map_options);
+
+        if (r < 0) {
+          std::cerr << "rbd: couldn't parse pool map options" << std::endl;
+          return r;
+        }
+
+        break;
+      }
+    }
+  }
```

Comment:
What is the purpose of this entire `if` statement?  It just iterates over the same `options` vector...

Reply:
Agreed, that needs to be removed.